Symfony/CS/Console/Output/ProcessOutput.php

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * Output writer to show the process of a FixCommand
+ *
+ * @internal
+ */
+class ProcessOutput
+{
+    /**
+     * File statuses map.
+     *
+     * @var array
+     */
+    private static $eventStatusMap = array(
+        FixerFileProcessedEvent::STATUS_UNKNOWN => array('symbol' => '?', 'format' => '%s', 'description' => 'unknown'),
+        FixerFileProcessedEvent::STATUS_INVALID  => array('symbol' => 'I', 'format' => '<bg=red>%s</bg=red>',  'description' => 'invalid file syntax, file ignored'),
+        FixerFileProcessedEvent::STATUS_SKIPPED => array('symbol' => '', 'format' => '<bg=cyan>%s<bg=cyan>', 'description' => ''),
+        FixerFileProcessedEvent::STATUS_NO_CHANGES => array('symbol' => '.', 'format' => '%s', 'description' => 'no changes'),
+        FixerFileProcessedEvent::STATUS_FIXED => array('symbol' => 'F', 'format' => '<bg=green>%s</bg=green>', 'description' => 'fixed'),
+        FixerFileProcessedEvent::STATUS_EXCEPTION => array('symbol' => 'E', 'format' => '<bg=red>%s</bg=red>', 'description' => 'error'),
+        FixerFileProcessedEvent::STATUS_LINT => array('symbol' => 'E', 'format' => '<bg=red>%s</bg=red>', 'description' => 'error'),
+    );
+
+    /**
+     * Stream output instance.
+     *
+     * @var StreamOutput
+     */
+    private $output;
+
+    /**
+     * Event dispatcher instance.
+     *
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * @param EventDispatcher $dispatcher
+     * @param StreamOutput    $output
+     */
+    public function __construct(EventDispatcher $dispatcher, StreamOutput $output)
+    {
+        $this->output = $output;
+        $this->eventDispatcher = $dispatcher;
+        $this->eventDispatcher->addListener(FixerFileProcessedEvent::NAME, array($this, 'onFixerFileProcessed'));
+    }
+
+    public function onFixerFileProcessed(FixerFileProcessedEvent $event)
+    {
+        $status = self::$eventStatusMap[$event->getStatus()];
+
+        if ($this->output->isDecorated()) {
+            $symbol = sprintf($status['format'], $status['symbol']);
+        } else {
+            $symbol = $status['symbol'];
+        }
+
+        $this->output->write($symbol);
+    }
+
+    public function printLegend()
+    {
+        $this->output->write("\nLegend:");
+        $symbols = array();
+
+        foreach (self::$eventStatusMap as $status) {
+            if ('' === ($status['symbol'])) {
+                continue;
+            }
+
+            if ($this->output->isDecorated()) {
+                $symbol = sprintf($status['format'], $status['symbol']);
+            } else {
+                $symbol = $status['symbol'];
+            }
+
+            if (in_array($symbol, $symbols, true)) {
+                continue;
+            }
+
+            $symbols[] = $symbol;
+            $this->output->write(sprintf(' %s-%s,', $symbol, $status['description']));
+        }
+
+        $this->output->write("\n");
+    }
+
+    public function close()
```

Comment:
I'll change this, sorry @GrahamCampbell I don know how I made your comments disappear, that shouldn't be possible on github (at least for a mortal like me ;) ) right?